Terraform Docs (#3174)
hashicorp/web-unified-docs: content/terraform-policy/v0.2.x (beta)/docs/policy/reference/policy/index.mdx

---
page_title: Policy Configuration Reference
description: >-
  Reference documentation for Terraform policy file syntax and configuration blocks.
# START AUTO GENERATED METADATA, DO NOT EDIT
created_at: 2026-07-14T11:44:47-05:00
last_modified: 2026-07-14T19:01:34.000Z
# END AUTO GENERATED METADATA
---

# Policy Configuration Reference

Terraform policy's syntax is based on HCL, and supports a number of block types that you can use when you write your policies. Unlike Terraform, Terraform policy evaluates each policy file in a given directory separately. We recommend that you include sets of policies that you want to enforce together in each policy file, and related policies in separate files in the same directory structure.

@include "beta.mdx"

## Policy block syntax

Terraform policy block syntax is based on HCL, and similar to Terraform's syntax.

```hcl
<block_type> "<type>" ["<label>"] {
  <attribute_name> = <attribute_value>
  ...

  <block_name> {
    <block_attribute_name> = <block_attribute_value>
    ...
  }
}
```

Each policy file can include:

> - An optional [`policy` block](/terraform/policy/reference/policy/policy) to configure Terraform policy itself.

Author: ```suggestion
- A [`policy` block](/terraform/policy/reference/policy/policy) to configure Terraform policy itself.
```
